src/psyclone/tests/psyad/domain/lfric/test_lfric_adjoint.py

```diff
@@ -57,81 +68,543 @@ def test_generate_lfric_adjoint_no_container_error(fortran_reader):
+    assert access.lower() == "gh_readwrite"
+    assert metadata.meta_args[4].access.lower() == access.lower()
+
+
+def test_update_access_metadata_inc():
+    '''Test that a discontinuous field with intent inout results in
+    metadata with access gh_readwrite.
+
+    '''
+    metadata, dummy, field_1, field_2, field_3, scalar, operator = \
+        get_metadata_args()
+    assert metadata.meta_args[1].access.lower() == "gh_read"
+    arguments = [
+        dummy, dummy, field_1, field_2, field_3, scalar, dummy, operator]
+    access = _update_access_metadata("field_2", arguments, metadata)
+    assert access.lower() == "gh_readwrite"
+    assert metadata.meta_args[1].access.lower() == access.lower()
+
+
+def test_update_access_metadata_field_inc():
+    '''Test that a continuous field with intent inout results in metadata with
+    access gh_inc.
+
+    '''
+    metadata, dummy, field_1, field_2, field_3, scalar, operator = \
+        get_metadata_args()
+    assert metadata.meta_args[2].access.lower() == "gh_read"
+    arguments = [
+        dummy, dummy, field_1, field_2, field_3, scalar, dummy, operator]
+    access = _update_access_metadata("field_3", arguments, metadata)
+    assert access.lower() == "gh_inc"
+    assert metadata.meta_args[2].access.lower() == access.lower()
+
+
+def test_update_access_metadata_metaarg(monkeypatch):
+    '''Test that the _update_access_metadata method raises the expected
+    exception when an unsupported metaarg is found. We need to
+    monkeypatch to force this exception.
+
+    '''
+    metadata, dummy, field_1, field_2, _1, _2, _3 = \
+        get_metadata_args()
+    monkeypatch.setattr(metadata._meta_args, "_meta_args_args",
+                        [metadata.meta_args[0], None])
+    # Avoid validity checks
+    monkeypatch.setattr(
+        ArgIndexToMetadataIndex, "mapping", lambda _: {1: 0, 2: 1})
+    arguments = [dummy, field_1, field_2]
+    with pytest.raises(InternalError) as info:
+        _ = _update_access_metadata("field_2", arguments, metadata)
+    assert "Found unexpected meta arg class 'NoneType'." in str(info.value)
+
+
+def test_update_access_metadata_write(monkeypatch):
+    '''Test that a field with intent out results in metadata with access
+    gh_write. We monkeypatch the field to create this case.
+
+    '''
+    metadata, dummy, field_1, field_2, field_3, scalar, operator = \
+        get_metadata_args()
+    assert metadata.meta_args[1].access.lower() == "gh_read"
+    monkeypatch.setattr(
+        field_2.interface, "_access", ArgumentInterface.Access.WRITE)
+    arguments = [
+        dummy, dummy, field_1, field_2, field_3, scalar, dummy, operator]
+    access = _update_access_metadata("field_2", arguments, metadata)
+    assert access.lower() == "gh_write"
+    assert metadata.meta_args[1].access.lower() == access.lower()
+
+
+def test_update_access_metadata_read(monkeypatch):
+    '''Test that a field with intent out results in metadata with access
```

Comment:
"intent in"?

Reply:
No, that is correct, it is just that the docstring is badly written. I've hopefully improved it enough.